File: 2022/day10/cathode.cpp
#include <iostream>
#include <fstream>
#include <list>
#include <cstring>
using namespace std;
class CPU {
    public:
        int cycleCount;
        CPU(list<int> pl) {
            X = 1;
            clock = 0;
            printList = pl;
            cycleCount = 0;
        }
        void exec(string inst) {
            if(inst.starts_with("noop")) {
                cycle();
            }
            else {
                cycle();
                cycle();
                int val = stoi(inst.substr(5));
                X+=val;
            }
        }
    private:
        int X;
        int clock;
        list<int> printList;
        void cycle() {
            clock+=1;
            if(find(printList.begin(),printList.end(),clock) != printList.end()) {
                cycleCount+=X*clock;
            }
        }
};
int main(int argc, char *argv[]) {
    ifstream input("input.txt");
    string fileText;
    int cnt = 0;
    list<int> printCycles = {20,60,100,140,180,220};
    CPU c = CPU(printCycles);
    while(getline(input, fileText)) {
        c.exec(fileText);
        cnt+=1;
    }
    cout << c.cycleCount;
    cout << "\n";
}

File: 2022/day10/cathode.py
def parse():
    with open("input.txt","r") as f:
        data = f.read().split("\n")
    return data
data = parse()
reg = 1
clock = 0
mult = 0
for inst in data:
    split = inst.split(" ")
    if(split[0]=="noop"):
        clock+=1
        if(clock in [20, 60, 100, 140, 180, 220]):
            print(reg)
            mult+=clock*reg
    elif(split[0]=="addx"):
        clock+=1
        if(clock in [20, 60, 100, 140, 180, 220]):
            print(reg)
            mult+=clock*reg
        clock+=1
        if(clock in [20, 60, 100, 140, 180, 220]):
            print(reg)
            mult+=clock*reg
        reg+=int(split[1])
print(mult)